cmake_scripts/FindArrow.cmake

```diff
@@ -106,10 +106,10 @@ if (ARROW_INCLUDE_DIR AND ARROW_LIBS)
     set(ARROW_PYTHON_SHARED_IMP_LIB ${ARROW_LIBS}/${ARROW_PYTHON_LIB_NAME}.lib)
   else()
     set(ARROW_STATIC_LIB ${ARROW_PYTHON_LIB_PATH}/libarrow.a)
-    set(ARROW_PYTHON_STATIC_LIB ${ARROW_PYTHON_LIB_PATH}/libarrow_python.a)
+    set(ARROW_PYTHON_STATIC_LIB ${ARROW_PYTHON_LIB_PATH}/python3.8/site-packages/pyarrow/libarrow_python.a)
```

Comment:
We could but it will require running a python script from a cmake file

Reply:
But I suppose it should be possible to not hardcode the `python3.8` ? 

Above in this file, I see other places where `site-packages` is referenced without hardcoding this

Reply:
@jorisvandenbossche I made that adjustment, I'm not able to confirm the libarrow_python.a exists in that location

Reply:
As a possible alternative, and to basically do what @kou suggested, it might be worth to take a look at what FindNumpy.cmake does. 

Here in turbodobc, you seem to use `find_path` with a set of options:

https://github.com/blue-yonder/turbodbc/blob/ca36325a61cda7e1af628b7a947de8a3a00149be/cmake_scripts/Findnumpy.cmake#L8

In arrow, our FindNumpy.cmake actually does run a bit of python:

https://github.com/apache/arrow/blob/b21bf749f291367f85ad61751205e7deeee92bc7/cpp/cmake_modules/FindNumPy.cmake#L49-L50

Reply:
This is perfect! I'm going to clean everything up now

Reply:
pyarrow.get_library_dirs() was attempted before completely removing the static references from FindArrow. pyarrow.get_include() is used for the newly created include paths post pyarrow10 changes